### ISSUE TITLE
Fast kmeans++

### ISSUE THREAD

**Issue by franknoe** [markovmodel/PyEMMA#1003]
This paper suggests a k-means seeding method that is much faster than kmeans++ and nearly as good. The algorithm constructs a Markov chain on the data points, and in practice k-1 chains of length 100 seem enough to get approximately equal quality as kmeans++. Importantly, only one full pass over the data is needed.

https://papers.nips.cc/paper/6478-fast-and-provably-good-seedings-for-k-means.pdf

A python implementation is also available and can be installed with
    pip install kmc2

I bet the code is available too, but I haven't searched for it yet.

**Comment by psolsson** [comment]
Cython implementation is here:
https://github.com/obachem/kmc2

**Comment by marscher** [comment]
That is a nice improvement. We just need to substitute the metric calculations by ours and should be good to go. MIT license is permissive, right? So we just leave the copyright notes and and can include the code.

**Comment by franknoe** [comment]
Yes permissive. The change is probably trivial if we only want to work 
on data in memory, but this algorithm could also handle data that can't 
be fully loaded.

Am 06/12/16 um 15:05 schrieb Martin K. Scherer:
